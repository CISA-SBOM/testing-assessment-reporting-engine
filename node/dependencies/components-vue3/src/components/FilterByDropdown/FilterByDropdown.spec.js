import { describe, expect, it } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import Component from "./FilterByDropdown.vue";

describe("FilterByDropdown.vue", () => {
  it("is a Vue instance", () => {
    const wrapper = shallowMount(Component, {
      directives: {
        'uid': {
          created(el) {
            el.setAttribute('id', 'unique-id')
          }
        }
      }
    })
    expect(wrapper.vm).toBeTruthy();
  });

  it("matches snapshot with no props assigned", async () => {
    const props = {};
    const wrapper = shallowMount(Component, {
      props,
      directives: {
        'uid': {
          created(el) {
            el.setAttribute('id', 'unique-id')
          }
        }
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
