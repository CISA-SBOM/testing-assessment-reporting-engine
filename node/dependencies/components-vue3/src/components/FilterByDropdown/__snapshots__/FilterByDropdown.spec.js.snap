// Vitest Snapshot v1

exports[`FilterByDropdown.vue > matches snapshot with no props assigned 1`] = `
"<div class=\\"inline-block\\" data-id=\\"sds-filter-by-dropdown\\">
  <div class=\\"inline-block w-full\\"><button id=\\"unique-id\\" class=\\"link link-secondary\\" type=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\"><span>Filter</span><svg class=\\"inline-block self-center w-5 h-5 -mr-1\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\" fill=\\"currentColor\\">
        <path fill-rule=\\"evenodd\\" d=\\"M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z\\" clip-rule=\\"evenodd\\"></path>
      </svg></button></div>
  <!---->
</div>"
`;
